### ISSUE TITLE
Using Using Pod-based Workload Identity doesn't appear to work in GCP

### ISSUE THREAD

**Issue by kiamatt** [external-secrets/external-secrets#2255]
I'm using version 0.8.1 of the helm chart and following the docs set [here.](https://external-secrets.io/v0.8.1/provider/google-secrets-manager/)

I've tried making a new K8S service account with the correct annotation, and passing an annotation to the service account created by the operator. My annotation is `iam.gke.io/gcp-service-account: <my-gcp-service-account>@<my-project>.iam.gserviceaccount.com`.

I also pass the project ID to the secret store.

In either case, I get this error in the secret store:

` Warning  InvalidProviderConfig  10s (x14 over 51s)     secret-store  unable to get credentials: compute: Received 403 `Unable to generate access token; IAM returned 403 Forbidden: Permission 'iam.serviceAccounts.getAccessToken' denied on resource (or it may not exist).
This error could be caused by a missing IAM policy binding on the target IAM service account.`

The interesting this is that the docs make it sound like you don't need the `roles/iam.serviceAccountTokenCreator` role applied if you're using the Pod-based Workload Identity, but I have that role set anyway.

**Comment by kiamatt** [comment]
> Hey @kiamatt, did you add the `roles/secretmanager.secretAccessor` (or similar) ? We have e2e tests that cover GKE Workload Identity, [see setup here](https://github.com/external-secrets/external-secrets/blob/main/terraform/gcp/eso_gcp_modules/gke/main.tf#L1-L28). Maybe that helps with the set up.
> 
> Can you please share your SecretStore? That could help with debugging this issue.

@moolen  Thanks for the quick response; I do have that role set. I'll attach some images here.

Roles on the service account:
![image](https://user-images.githubusercontent.com/16911691/234028219-a56d2de6-a1c5-4724-a82d-048f0f7c88e9.png)

service account set on the ESO pod:
![image](https://user-images.githubusercontent.com/16911691/234028538-4fca9c4a-92fa-4a4a-b418-adabd6c4ff21.png)

Secret Store:
`apiVersion: external-secrets.io/v1beta1
kind: SecretStore
metadata:
  name: secretstore
  namespace: {{ .Values.target_namespace }}
spec:
  provider:
    gcpsm:
      projectID: {{ .Values.project_id }}`

Error on the Secret store:
![image](https://user-images.githubusercontent.com/16911691/234029099-c2f3709d-ae3e-4803-b251-e318c0013c6e.png)

**Comment by moolen** [comment]
The error says that there's probably an IAM policy binding missing. Did you double check that this exists?

referenced docs: https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#authenticating_to

**Comment by kiamatt** [comment]
> The error says that there's probably an IAM policy binding missing. Did you double check that this exists?
> 
> referenced docs: https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#authenticating_to

@moolen Unfortunately, I do have those bindings in place:
![image](https://user-images.githubusercontent.com/16911691/234090117-492aedec-4428-489e-9b50-62e57626d174.png)

I also have the annotation set on the service account, to use the GCP service account. I also made sure that the pod is running on a node with `iam.gke.io/gke-metadata-server-enabled` set to `true`.

**Comment by moolen** [comment]
The `roles/iam.workloadIdentityUser` does it point to the correct Kubernetes Kind=ServiceAccount in the right namespace?

```sh
$ gcloud iam service-accounts add-iam-policy-binding GSA_NAME@GSA_PROJECT.iam.gserviceaccount.com \
    --role roles/iam.workloadIdentityUser \
    --member "serviceAccount:PROJECT_ID.svc.id.goog[NAMESPACE/KSA_NAME]" # <-- this one here
```

**Comment by kiamatt** [comment]
@moolen Nope, I didn't have that set correctly! Thank you so, so much. And apologies, I missed that in the docs.
